### PR TITLE
Use the CreationTime to update LastModifiedTime 

### PR DESCRIPTION
**Pull request by mbaijal**
Issue #, if available:
Fixing [comment](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/pull/40#pullrequestreview-751856481) on previous PR #40 

Description of changes:
 - Use the CreationTime to update LastModifiedTime 
 - Add associated test to check LastModifiedTime

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py [hunk 1]
```diff
@@ -201,7 +216,10 @@ def test_update(self, applicationautoscaling_client, generate_sagemaker_policy):
         
         # Update the ScalableTarget
         target_spec["spec"]["maxCapacity"] = updatedMaxCapacity
+        assert "lastModifiedTime" in target_resource["status"]
+        last_modified_time = target_resource["status"]["lastModifiedTime"]
         k8s.patch_custom_resource(target_reference, target_spec)
+        assert self.wait_until_update(target_reference, last_modified_time) == True
         sleep(5)
```
Comment (surajkota) [inline test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:223]: no need of sleep now? here and line 239

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py [hunk 2]
```diff
@@ -227,8 +249,7 @@ def test_update(self, applicationautoscaling_client, generate_sagemaker_policy):
     def test_delete(self, applicationautoscaling_client, generate_sagemaker_policy):
         (
             resource_id,
-            target_spec,
-            target_reference,
+            target_reference, target_spec, target_resource,
```
Comment (surajkota) [inline test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:252]: nit: strange formatting

File: pkg/resource/scaling_policy/custom_api.go [hunk 3]
```diff
@@ -19,8 +19,13 @@ import (
 	"time"
 )
 
+// customSetOutputUpdate sets the LastModifiedTime field to the creationTime
+func (rm *resourceManager) customSetLastModifiedTimeToCreationTime(ko *svcapitypes.ScalingPolicy) {
+	ko.Status.LastModifiedTime = ko.Status.CreationTime
```
Comment (surajkota) [inline pkg/resource/scaling_policy/custom_api.go:24]: using this code directly in generator config should work but ok either way

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py [hunk 4]
```diff
@@ -127,6 +127,23 @@ def generate_sagemaker_policy(generate_sagemaker_target):
 @service_marker
 @pytest.mark.canary
 class TestSageMakerEndpointAutoscaling:
+    def wait_until_update(
+        self, reference, previous_modified_time, wait_period=30, wait_time=5
+    ):
+        for i in range(wait_period):
```
Comment (surajkota) [inline test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:133]: should we switch the wait_periods to 2 and wait_time to 30 or less

30 times every 5 sec seems unnecessary

**Review by surajkota — CHANGES_REQUESTED** [review]

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 1 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:223]: ++ will remove

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 2 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:252]: wasn't really ready for a review, was it :D

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 4 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:133]: sg

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 1 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:223]: done

File: pkg/resource/scaling_policy/custom_api.go (hunk 3 above)
Comment (mbaijal) [reply pkg/resource/scaling_policy/custom_api.go:24]: prefer the current code for modularity and cleanliness, so keeping as is.

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 4 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:133]: done.

File: test/e2e/tests/test_sagemaker_endpoint_autoscaling.py (hunk 2 above)
Comment (mbaijal) [reply test/e2e/tests/test_sagemaker_endpoint_autoscaling.py:252]: Ran black.

File: pkg/resource/scalable_target/custom_api.go [hunk 5]
```diff
@@ -35,8 +35,15 @@ func (rm *resourceManager) customDescribeScalableTarget(
 	}
 }
 
+// customSetOutputUpdate sets the LastModifiedTime field to the creationTime
```
Comment (surajkota) [inline pkg/resource/scalable_target/custom_api.go:38]: nit: method name not matching in comment, same for other

and in future lets just rename this file to hooks.go

**Review by surajkota — APPROVED** [review]
/approve

File: pkg/resource/scalable_target/custom_api.go (hunk 5 above)
Comment (mbaijal) [reply pkg/resource/scalable_target/custom_api.go]: fixed the comments, we can change the filename in another PR.

**Review by surajkota — APPROVED** [review]
/approve
/lgtm

**Pull request merged**